internal/venice-client-common/src/main/java/com/linkedin/venice/compression/GzipPool.java

```diff
@@ -0,0 +1,51 @@
+package com.linkedin.venice.compression;
+
+import com.linkedin.venice.utils.concurrent.CloseableThreadLocal;
+import java.io.IOException;
+import java.io.OutputStream;
+
+
+abstract class GzipPool {
+  private static class ReusableObjects implements AutoCloseable {
+    final SettableOutputStream stream = new SettableOutputStream();
+    final ReusableGzipOutputStream gzipOutputStream = new ReusableGzipOutputStream(stream);
+
+    @Override
+    public void close() throws Exception {
+      stream.close();
+    }
```

Comment:
Can we close the `gzipOutputStream` also?

Reply:
sure